bodhi/tests/server/services/test_updates.py

```diff
@@ -2565,6 +2565,60 @@ def test_edit_testing_update_with_new_builds(self, publish, *args):
         self.assertEquals(len(publish.call_args_list), 3)
         publish.assert_called_with(topic='update.edit', msg=ANY)
 
+    @mock.patch(**mock_valid_requirements)
+    @mock.patch('bodhi.server.notifications.publish')
+    def test_edit_testing_update_with_failed_title_change(self, publish, *args):
+        """Test that editing a multi-build update is still possible if title updating failed.
+
+        At some point, the text of the title was used to determine whether a build was
+        in the current update.
+        """
+        nvrs = 'a-1.0-1.fc17,b-1.0-1.fc17,c-1.0-1.fc17'
+        args = self.get_update(nvrs)
+        r = self.app.post_json('/updates/', args)
+        publish.assert_called_with(topic='update.request.testing', msg=ANY)
+
+        # Mark it as testing
+        upd = Update.get(r.json_body['alias'])
+        newtitle = upd.title.replace('a-1.0', 'a-1.5')
+        upd.title = newtitle
+        upd.status = UpdateStatus.testing
+        upd.request = None
+        self.db.commit()
+
+        args['edited'] = newtitle
+        # Kept a and c the same, bumped b
+        args['builds'] = ['a-1.0-1.fc17', 'b-2.0-1.fc17', 'c-1.0-1.fc17']
+        r = self.app.post_json('/updates/', args)
+        up = r.json_body
+        self.assertEquals(up['title'], u'a-1.0-1.fc17 b-2.0-1.fc17 c-1.0-1.fc17')
+        self.assertEquals(up['status'], u'pending')
+        self.assertEquals(up['request'], u'testing')
+        self.assertEquals(up['comments'][-1]['text'],
+                          u'This update has been submitted for testing by guest. ')
+        comment = textwrap.dedent("""
+        guest edited this update.
+
+        New build(s):
+
+        - b-2.0-1.fc17
+
+        Removed build(s):
+
+        - b-1.0-1.fc17
```

Comment:
One day I will add tags to the Comment model so we don't have to have Bodhi test its own comments to see if stuff like this happened :)